### PR TITLE
feat(progress): add progress component

### PR DESCRIPTION
**Pull request by rav3**
## :open_book: Description

Fixes # (issue)

## :pencil2: Type of change

```
Please delete options that are not relevant.
```

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## :test_tube: How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## :white_check_mark: Checklist:

- [x] My code follows the SGDS style guidelines and naming conventions
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/Progress/sgds-progress-bar.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        aria-valuemin=${this.ariamin}
+        aria-valuemax=${this.ariamax}
+      >
+        ${this.label ? html`${this.label}` : undefined}
+      </div>
+    `;
+  }
+}
+
+export type progressbarVariant = "primary" | "secondary" | "success" | "danger" | "warning" | "info" | "light" | "dark";
```
Comment (clukhei) [inline src/components/Progress/sgds-progress-bar.ts:73]: type syntax should be pascal case ProgressBarVariant

File: src/components/Progress/sgds-progress.scss [hunk 2]
```diff
@@ -0,0 +1,8 @@
+:host{
+    --sgds-progress-height: 1rem;
```
Comment (clukhei) [inline src/components/Progress/sgds-progress.scss:2]: docs in convention mention that `--componentname-subcomponentname-element-state-properties` 
no need sgds prefix

File: src/components/Progress/sgds-progress.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+export class SgdsProgress extends SgdsElement {
+  static styles = [SgdsElement.styles, styles];
+
+  /** Forwarded to the base wrapper of sgds-progress. Can be used to insert any utility classes such as `me-auto` */
+  @property({ type: String, reflect: true }) progressClasses: string;
+
+  render() {
+    return html`
+      <div
+        part="base"
```
Comment (clukhei) [inline src/components/Progress/sgds-progress.ts:23]: this csspart not documented. if not needed can remove?

**Review by clukhei — COMMENTED** [review]
what will happen if you remove sgds-progress and just use <sgds-progress-bar> ?

**Comment by rav3** [comment]
> what will happen if you remove sgds-progress and just use ?

These will not work if sgds-progress wrapper is absent:
- progress bar height
- the grey background color of the wrapper
- rounded border

File: src/components/Progress/sgds-progress-bar.ts (hunk 1 above)
Comment (rav3) [reply src/components/Progress/sgds-progress-bar.ts]: resolved in [d209e8e](https://github.com/GovTechSG/sgds-web-component/pull/110/commits/d209e8e67ce3164679959bc0a52023a3697c4820)

File: src/components/Progress/sgds-progress.scss (hunk 2 above)
Comment (rav3) [reply src/components/Progress/sgds-progress.scss]: resolved in [a9b34cc](https://github.com/GovTechSG/sgds-web-component/pull/110/commits/a9b34ccc04c97965cc602a23aac0a06bce3fbb86)

File: src/components/Progress/sgds-progress.ts (hunk 3 above)
Comment (rav3) [reply src/components/Progress/sgds-progress.ts]: resolved in [a9b34cc](https://github.com/GovTechSG/sgds-web-component/pull/110/commits/a9b34ccc04c97965cc602a23aac0a06bce3fbb86)

File: src/components/Progress/sgds-progress-bar.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import { property } from "lit/decorators.js";
+import { styleMap } from "lit/directives/style-map.js";
+import { classMap } from "lit/directives/class-map.js";
+import SgdsElement from "../../base/sgds-element";
+import styles from "./sgds-progress-bar.scss";
+import { html } from "lit";
+
+/**
+ * @csspart label - Progress bar label
```
Comment (clukhei) [inline src/components/Progress/sgds-progress-bar.ts:9]: there is no part=label in the html below

**Review by clukhei — APPROVED** [review]
theres an extra(?) csspart label with not corresponding part=label in the html

File: src/components/Progress/sgds-progress-bar.ts (hunk 4 above)
Comment (rav3) [reply src/components/Progress/sgds-progress-bar.ts]: resolved in [bc37b42](https://github.com/GovTechSG/sgds-web-component/pull/110/commits/bc37b42418386e8c82ebece48dbf312fda680f2e)

**Review by clukhei — APPROVED** [review]

**Pull request merged**